b2sdk/scan/folder.py

```diff
@@ -204,6 +232,32 @@ def _walk_relative_paths(
         #
         # This is because in Unicode '.' comes before '/', which comes before '0'.
         names = []  # list of (name, local_path, relative_file_path)
+
+        if starting_dir is None:
+            starting_dir = local_dir
+
+        # Set max_symlink_visits based on the policies_manager object
+        # Fall back to 10 if max_symlink_visits is not set
+        # FIXME should we use a default value from somewhere else?
+        max_symlink_visits = getattr(policies_manager, 'max_symlink_visits', 10)
+
+        if visited_symlinks is None:
+            visited_symlinks = {}
+
+        if os.path.islink(local_dir):
+            absolute_path = os.path.abspath(local_dir)
+            real_path = os.readlink(absolute_path)
+
+            if is_ancestor_directory(local_dir, starting_dir):
+                # Skip symlink if it points to an ancestor directory of the initial scan directory
+                return
+
+            symlink_count = visited_symlinks.get(real_path, 0)
```

Comment:
this wouldn't be needed if `visited_symlinks` would be `defaultdict(int)` or `collections.Counter` (the former seems more appropriate here)